river/cluster/clustream.py

```diff
@@ -0,0 +1,363 @@
+      about the data that arrived within the period (T-h,T)
+
+    max_kernels
+      The Maximum number of micro kernels to use
+
+    kernel_radius_factor
+      Multiplier for the kernel radius
+      When deciding to add a new data point to a micro-cluster, the maximum boundary
+      is defined as a factor of the kernel_radius_factor of the RMS deviation of the
+      data points in the micro-cluster from the centroid
+
+    number_of_clusters
+        the clusters returned by the Kmeans algorithm using the summaries statistics
+
+    kwargs
+        Other parameters passed to the incremental kmeans at `river.cluster.KMeans`
+
+    Attributes
+    ----------
+    centers : dict
+        Central positions of each cluster.
+
+    Examples
+    --------
+
+    In the following implementation, due to the limited number of points passed through the model,
+    max_kernels and time_window are set relatively low. Moreover, all points are learnt before any predictions are made.
+    The halflife are set at 0.4, passing from cluster.KMeans, to test the functionality of keyword arguments.
+
+    >>> from river import cluster
+    >>> from river import stream
+
+    >>> X = [
+    ...     [1, 2],
+    ...     [1, 4],
+    ...     [1, 0],
+    ...     [4, 2],
+    ...     [4, 4],
+    ...     [4, 0]
+    ... ]
+
+    >>> clustream_test = cluster.Clustream(time_window = 1, max_kernels = 3, number_of_clusters = 2, seed = 0, halflife = 0.4)
+
+    >>> for i, (x, _) in enumerate(stream.iter_array(X)):
+    ...     clustream_test = clustream_test.learn_one(x)
+
+    >>> clustream_test.predict_one({0: 1, 1: 1})
+    1
+
+    >>> clustream_test.predict_one({0: 4, 1: 3})
+    0
+
+    References
+    ----------
+    .. [1] A. Kumar, A. Singh, and R. Singh, 2017. An efficient hybrid-clustream algorithm
+       for stream mining. In Proceedings of the 13th International Conference on Signal-Image Technology & Internet-Based System (SITIS).
+       DOI: 10.1109/SITIS.2017.77
+
+    """
+
+    def __init__(
+        self,
+        seed: int = None,
+        time_window: int = 1000,
+        max_kernels: int = 100,
+        kernel_radius_factor: int = 2,
+        number_of_clusters: int = 5,
+        **kwargs
+    ):
+        super().__init__()
+        self.time_window = time_window
+        self.time_stamp = -1
+        self.kernels = {n: None for n in range(max_kernels)}
+        self.initialized = False
+        self.buffer = {}
+        self.buffer_size = max_kernels
+        self.centers = (
+            {}
+        )  # add this to retrieve centers later for the evaluation of models through
+        self.kernel_radius_factor = kernel_radius_factor
+        self.max_kernels = max_kernels
+        self.number_of_clusters = number_of_clusters
+        self._train_weight_seen_by_model = 0.0
+        self.seed = seed
+        self.kwargs = kwargs
+
+    def learn_one(self, x, sample_weight=None):
+        """Incrementally trains the model.
```

Comment:
If I were you I would put all this description in the class docstring and leave this docstring empty. I think it's better to describe the model in one place. If you don't specify the `learn_one` docstring, then the docstring from the `Clusterer` class will be inherited, which should be good enough.

Reply:
As I believe, each `learn_one` function of each algorithm is fairly different from each other, so it's kind of necessary to specify what this function does for each algorithm (although the purpose is the same, which is "incrementally trains the model").